bin/dothething

```diff
@@ -30,147 +34,247 @@ rescue OptionParser::InvalidOption => e
  exit 1
 end.parse!
 
-class State
- #Loads the config file
- def initialize(state)
-  @statefile = state 
-  @sf = YAML.load_file(@statefile)
- end
+#If no command was entered display help and exit 
+if options.empty?
+ puts opt_parser
+ exit 1
+end
 
- #Toggles value in state hash
- def toggle(name)
-  #Finds all true/false values in hash
-  t = @sf[name].key(true)
-  f = @sf[name].key(false) 
+module ConfigFile
+ class Config
+  #Load the config file
+  def initialize
+   configfile = File.join($path, 'config.yml')
+   @config = YAML.load_file(configfile)
+  end
+  
+  #Check if a setting has been set
+  def setting_exists?(setting)
+   if @config[setting] != nil then
+    return true 
+   else
+    puts "The config file does not contain a path for the setting: #{setting}"
+    exit 1 
+   end 
+  end
+  
+  #Set the directory for the job   
+  def jobdir(name)
+   setting_exists?('jobdir')
+   @jobdir = @config['jobdir']
+   
+   #Check if the path given in the file is absolute or relative   
+   if @jobdir.include? "../"
```

Comment:
Don't think this is a good method to differentiate between an absolute or relative path in the config file, consider if I just put `jobdir: jobs/` in the config file, thats still a relative path. Instead I suggest just confirming if the raw string can be located with `::File::exist?` and if not assume its a full path..

Reply:
Reworked this so that it checks if the path available exists, returning an absolute/relative path as needed